### PR TITLE
fix(antigravity): suppress duplicated pool quotas without losing distinct rows

### PR DESCRIPTION
**Pull request by hhh2210**
Remote Antigravity responses can repeat a consumed pool under internal image, lite, and autocomplete model IDs. A pool at 70% remaining and variants with the same known fraction and concrete reset now produce one pool summary. Local rows, unknown usage, different reset windows, and independently differing fractions remain visible when eligible.

Repeated canonical model IDs now select known usage before applying visibility rules. This prevents a reset-only copy from replacing a real quota, including when a fully unused known quota should hide the extra row. Extra rows retain their existing canonical IDs and compact-fallback identity; equal display titles do not merge independent models.

The reset-only pool list already has Gemini-then-Claude ordering. Removing its redundant sort and unused helpers keeps the complete fix at **8 fewer production lines**. Provider documentation and the 0.60.1 Unreleased changelog are updated.

Validation:
- Baseline regression fixtures reproduced duplicate remote rows and unknown-over-known canonical selection.
- Focused Antigravity and architecture checks: 453 tests plus 18 tests passed across the test products.
- `make check`: passed, zero SwiftLint violations in 2,217 files.
- Independent review through P2: no accepted/actionable findings.
- Full `make test`: 1,099 selections / 92 groups passed, zero failures or retries. Exact-head [CI passed](https://github.com/steipete/CodexBar/actions/runs/34697858177).

Tests use synthetic projection fixtures, including reversed inputs, exact versus nearly equal fractions, missing resets, both sources, fully unused known observations, family-specific autocomplete, stable IDs as usage changes, and reset-only ordering. No provider credentials or live account access were used for this maintainer validation.

Thanks @hhh2210 for the report, original fix, and contributed test coverage. Related context: #1139, #1209, #1509, #1580.

### CODE REVIEW

**Comment by hhh2210** [comment]
@clawsweeper re-review

**Comment by hhh2210** [comment]
@clawsweeper re-review

**Comment by hhh2210** [comment]
@clawsweeper re-review

**Comment by hhh2210** [comment]
@clawsweeper re-review

File: Sources/CodexBarCore/Providers/Antigravity/AntigravityStatusProbe.swift [hunk 1]
```diff
@@ -708,14 +731,74 @@ public struct AntigravityStatusSnapshot: Sendable {
...
+        _ rhs: AntigravityNormalizedModel,
+        compactFallbackModelID: String?) -> Bool
+    {
+        if let compactFallbackModelID {
+            if lhs.quota.modelId == compactFallbackModelID, rhs.quota.modelId != compactFallbackModelID {
+                return true
+            }
+            if rhs.quota.modelId == compactFallbackModelID, lhs.quota.modelId != compactFallbackModelID {
+                return false
+            }
```
[the remark this answers is not in the document]
Comment (hhh2210) [reply Sources/CodexBarCore/Providers/Antigravity/AntigravityStatusProbe.swift]: Fixed in df20306ef. Compact fallback identity is now preserved by the deduplicated group independently of the selected quota data. The 50%-remaining selectable fallback + 10%-remaining image variant now yields a fallback-prefixed row with 10% remaining and the image variant’s reset metadata. The new parameterized regression covers both input orders; the focused Antigravity and architecture runs passed 476 tests. Proof: https://github.com/hhh2210/CodexBar/blob/df20306ef09ea73a0d2bffa1bdd3b2dd62b7b7bd/.github/pr-proof/antigravity-pool-deduplication.log

File: Sources/CodexBarCore/Providers/Antigravity/AntigravityStatusProbe.swift [hunk 2]
```diff
@@ -669,19 +685,26 @@ public struct AntigravityStatusSnapshot: Sendable {
...
-                group.min { lhs, rhs in
-                    if lhs.quota.remainingPercent != rhs.quota.remainingPercent {
-                        return lhs.quota.remainingPercent < rhs.quota.remainingPercent
-                    }
-                    return lhs.quota.label < rhs.quota.label
-                }
+                // Collapse multiple wire IDs or variants formatting to the same title
+                // (e.g. duplicate "Gemini 3.1 Flash Lite" entries). Keep the most constrained
+                // (lowest remaining) to avoid duplicate windows while preserving compact fallback.
+                group.min(by: { Self.extraModelPrecedes($0, $1, compactFallbackModelID: compactFallbackModelID) })
```
[the remark this answers is not in the document]
Comment (hhh2210) [reply Sources/CodexBarCore/Providers/Antigravity/AntigravityStatusProbe.swift]: Fixed in df20306ef. Display-title deduplication now includes an observed reset timestamp; different timestamps, one missing timestamp, and both missing timestamps retain separate canonical model rows. Canonical-ID deduplication remains first, so this does not introduce duplicate downstream IDs. All three reset-boundary cases pass in the new parameterized test, alongside the existing same-reset collapse regression. Proof: https://github.com/hhh2210/CodexBar/blob/df20306ef09ea73a0d2bffa1bdd3b2dd62b7b7bd/.github/pr-proof/antigravity-pool-deduplication.log

**Comment by hhh2210** [comment]
@clawsweeper re-review

Updated to df20306ef with the two latest Codex findings addressed: quota selection preserves the most-constrained data independently of compact fallback identity, and title deduplication retains distinct/missing reset windows. The CI failure was stale source anchors in ProviderArchitectureGatekeeperTests; all seven anchors are refreshed without changing the allowlist reasons or fingerprints.

Validation is complete: 476 focused tests passed, the full `make test` passed all 1,090 selections in 91 groups on the first attempt (zero retries/timeouts), and `make check` passed with the documented host-only ambient TypeScript isolation. The PR body links the committed proof and retains the prior live-account evidence. Remote CI is still running; please reassess the current implementation and proof.

**Comment by steipete** [comment]
Addressed the earlier review findings in c29b9cb2f70. The final change keeps canonical model grouping and removes display-title grouping entirely, so independent model IDs, reset windows, and compact-fallback identity remain intact.

Known canonical observations now win before visibility filtering, including a fully unused quota; reset-only copies cannot replace them. Mirror filtering requires an exact known fraction and concrete matching reset, applies only to remote rows, and keeps represented-pool tracking for both sources. Autocomplete uses its recognized family when present and otherwise compares against Gemini. Comparisons remain unrounded.

The focused Antigravity and architecture runs passed (453 tests plus 18 tests across products), `make check` passed, and independent review through P2 found no actionable findings. Full local tests and exact-head CI are still running. These changes cover the ten existing outdated review threads; I am resolving those threads while retaining any new review findings for separate assessment.

**Comment by steipete** [comment]
Landed in a8704e025f61bd250723e049985793e9fb0a554b. Thank you @hhh2210.

The final fix suppresses only remote variants matching the represented pool's exact known fraction and concrete reset. Independent/local rows and canonical visibility IDs remain intact. Canonical grouping now prefers known usage before visibility filtering, including fully unused known quotas. The redundant reset-only pool sort and two helpers were removed: **8 fewer production lines**.

Proof:
- Baseline `swift test --filter AntigravityPoolDeduplicationTests` reproduced duplicate remote rows and unknown-over-known selection after the existing Sparkle test-runtime recovery.
- `swift test --filter 'Antigravity|ProviderArchitectureGatekeeperTests'`: 453 tests plus 18 tests passed across the test products.
- `make check`: passed, zero lint violations in 2,217 files.
- `make test`: 1,099 selections across 92 groups; all passed on the first attempt, no retries or timeouts.
- Tests ran with a synthetic HOME/CFFIXED_USER_HOME and the standard `SWIFT_TESTING`, Keychain-suppression, Codex-file-isolation, and session-file-isolation flags. Projection fixtures cover both sources, both input orders, exact and nearly equal fractions, known/unknown/full quotas, missing resets, family-specific autocomplete, identity stability, and pool ordering.
- Independent review through P2 found no accepted/actionable findings. [Exact-head CI](https://github.com/steipete/CodexBar/actions/runs/34697858177) passed for c29b9cb2f70444e43276b232822977352b0d3fc3; prior review threads were addressed before merge.

The maintainer proof uses synthetic projection data. The separate agy 1.2.2 authentication report #3586 remains open. Provider docs and the 0.60.1 Unreleased changelog include this fix; main is synchronized after landing.

**Comment by hhh2210** [comment]
Thanks for taking this through landing and tightening the projection. Keeping canonical model IDs and removing display-title grouping is a better boundary than my original approach, particularly for saved visibility and independent quota lanes. I also agree with selecting known canonical observations before visibility filtering. The separate authentication report remains a different issue.
